Source/MetroXHunter/Reload/ReloadComponent.h

```diff
@@ -0,0 +1,161 @@
+	void InitializeReloadData( UReloadData* NewReloadData );
+
+	UFUNCTION( BlueprintCallable, Category = "Reload" )
+	void UpdateAmmoCount( int NewCount );
+
+	UFUNCTION( BlueprintCallable, Category = "Reload" )
+	void ComputeReloadAmmoCount( int& NewMagazineAmmoCount, int& InventoryAmmoConsumed );
+
+	UFUNCTION( BlueprintCallable, Category = "Ammo" )
+	void DecrementAmmo();
+
+	UFUNCTION( BlueprintCallable, Category = "Reload" )
+	void FinalizeReload( int NewAmmoCount, float FinalWaitingTime, int InventoryAmmoCountUsed );
+
+	UFUNCTION( BlueprintCallable, Category = "Reload" )
+	bool IsGunFireLocked() const;
+
+	UFUNCTION( BlueprintCallable, Category = "Reload" )
+	bool IsAmmoFull() const;
+
+	UFUNCTION( BlueprintCallable, Category = "Reload" )
+	bool IsGunEmpty() const;
+
+	UFUNCTION( BlueprintCallable, Category = "Reload" )
+	void RetrieveReferences();
+
+	UFUNCTION( BlueprintCallable, Category = "Reload" )
+	void GetAmmoData( int& CurrentAmmo, int& MaxAmmo ) const;
+
+	UFUNCTION( BlueprintCallable, Category = "Reload" )
+	void GetNormalizedReloadTimings( float& PerfectReloadStartTime, float& ActiveReloadStartTime, float& ActiveReloadEndTime ) const;
+
+	UFUNCTION( BlueprintCallable, Category = "Reload" )
+	float GetNormalizedReloadElapsedTime() const;
+
+public:
+
+	UPROPERTY( EditAnywhere, BlueprintReadOnly, Category = "Reload|Inputs" )
+	UInputAction* ReloadAction = nullptr;
+
+	UPROPERTY( EditAnywhere, BlueprintReadOnly, Category = "Reload|Ammo" )
+	int CurrentAmmoInMagazine = 6;
+
+	UPROPERTY( EditAnywhere, BlueprintReadOnly, Category = "Reload|Ammo" )
+	int MaxAmmoInMagazine = 6;
+
+	UPROPERTY( BlueprintReadWrite, Category = "Reload" )
+	float ReloadElapsedTime = 0.0f;
+
+public:
+
+	DECLARE_DYNAMIC_MULTICAST_DELEGATE( FOnAmmoCountUpdated );
+	UPROPERTY( BlueprintAssignable, Category = "Reload|Events" )
+	FOnAmmoCountUpdated OnAmmoCountUpdated;
+
+	DECLARE_DYNAMIC_MULTICAST_DELEGATE( FOnUpdateCursorPosition );
+	UPROPERTY( BlueprintAssignable, Category = "Reload|Events" )
+	FOnUpdateCursorPosition OnUpdateCursorPosition;
+
+	DECLARE_DYNAMIC_MULTICAST_DELEGATE_OneParam( FOnReloadStateChanged, EReloadState, ReloadType );
+	UPROPERTY( BlueprintAssignable, Category = "Reload|Events" )
+	FOnReloadStateChanged OnReloadStateChanged;
+
+private:
+
+	UPROPERTY( EditAnywhere, BlueprintReadOnly, Category = "Reload|Enum", meta = ( AllowPrivateAccess = "true" ) )
+	EGunState CurrentGunState = EGunState::Idle;
+
+	UPROPERTY( EditAnywhere, BlueprintReadOnly, Category = "Reload|Data Asset", meta = ( AllowPrivateAccess = "true" ) )
+	UReloadData* ReloadDataAsset = nullptr;
+
+	UPROPERTY( BlueprintReadOnly, meta = ( AllowPrivateAccess = "true" ))
+	UInventoryComponent* PlayerInventory = nullptr;
+
+	UPROPERTY( meta = ( AllowPrivateAccess = "true" ) )
+	bool bIsReloadActive = false;
+
+	UPROPERTY( EditAnywhere, BlueprintReadOnly, Category = "Reload|Debug", meta = ( AllowPrivateAccess = "true" ) )
+	bool bUseInfiniteAmmo = false;
+
+private:
```

Comment:
Move private functions above private variables (and merge the two variable sections)